tests/integration/pilot/gateway_conformance_test.go

```diff
@@ -79,6 +86,48 @@ func TestGatewayConformance(t *testing.T) {
 		Run(func(ctx framework.TestContext) {
 			DeployGatewayAPICRD(ctx)
 
+			// Precreate the GatewayConformance namespaces, and apply the Image Pull Secret to them.
+			if ctx.Settings().Image.PullSecret != "" {
+				for _, ns := range conformanceNamespaces {
+					namespace.Claim(ctx, namespace.Config{
+						Prefix: ns,
+						Inject: false,
+					})
+					// Cannot apply the existing secret directly; need to sanitize the
+					// object metadata. Parse it to a k8s struct, clean the metadata,
+					// then convert back to yaml string so we can apply it.
+					// After that, patch the 'default' ServiceAccount in each NS
+					// to use the secret.
+					data, err := os.ReadFile(ctx.Settings().Image.PullSecret)
+					if err != nil {
+						t.Fatal(err)
+					}
+					secret := &v1.Secret{}
+					if err := k8syaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), len(data)).Decode(secret); err != nil {
+						t.Fatalf("Error unmarshalling yaml into secret: %s", err)
+					}
+					n := secret.ObjectMeta.Name
+					secret.ObjectMeta.Reset()
+					secret.ObjectMeta.SetName(n)
+					yml, err := yaml.Marshal(secret)
+					if err != nil {
+						t.Fatalf("Error marshalling secret: %s", err)
+					}
+					ctx.ConfigKube().YAML(ns, string(yml)).ApplyOrFail(t, apply.NoCleanup)
```

Comment:
can we do this in `claimKube`?

Reply:
Maybe with a bool option if we need it

Reply:
Yeah, let me rework it.

Reply:
That wound up being a much cleaner fix, good call.